library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/impl/ResourceLoaderImpl.java

```diff
@@ -523,4 +537,50 @@ public static void appendLanguageEntries(@NotNull Map<String, String> map) {
 			}
 		}
 	}
+
+	/* Static pack stuff */
+	private static List<ResourcePack> findUserStaticPacks() {
+		List<ResourcePack> returnList = new ArrayList<>();
+		try (Stream<Path> walkStream = Files.walk(QuiltLoader.getGameDir().resolve(STATIC_PACK_ROOT), 1)) {
+			for (Path path : walkStream.toList()) {
+				File pathAsFile = path.toFile();
+				String n = calcUserspacePackName(pathAsFile);
+				if (pathAsFile.isFile()) {
+					if (pathAsFile.toPath().toString().endsWith(".zip")) {
+						returnList.add(new ZipResourcePack(n, pathAsFile, false));
+					} else {
+						// Implementation detail: ._* and (some) *nix-derivative OSes (macOS, Linux distros, etc)
+						// ._* is a rare *nix filesystem helper file created to store file information normally placed in an extended attribute
+						// on HFS+ (Apple Native, typically found under macOS) or UFS (other *nix OSes) filesystems when interfacing with a filesystem
+						// that does NOT support such extended attributes (for instance, a FAT32 drive). Since these can have names of arbitrary lengths,
+						// the only way to detect them is via the shared prefix.
+						// Source: https://apple.stackexchange.com/a/14981
+						boolean fsHelper = pathAsFile.getName().startsWith("._");
+
+						if (!fsHelper) {
+							for (String fs : STATIC_USERSPACE_IGNORED_FILES) {
+								if (fs.equalsIgnoreCase(pathAsFile.getName())) {
+									fsHelper = true;
+									break;
+								}
+							}
+						}
+						if (!fsHelper) {
```

Comment:
```suggestion
```

Reply:
Should not be removed. The for loop just above this is able to set `fsHelper` to true and break out of the loop when it does so—removing the extra `if (!fsHelper)` check would cause the loose file log error to be displayed whenever the for loop is executed at all, not when the for loop never sets `fsHelper` to true. There's probably a more elegant way to write this segment, but this works while removing the second check does not.

Reply:
Then the ignored file list can be all lower case and change the condition to `pathAsFile.getName().starts with("._") || STATIC-USERSPACE_IGNORED_FILES.contains(pathAsFile.getName().toLowercase())`.

Reply:
Reimplemented in a very similar way to what you suggested as of latest commit, and so no longer uses a for loop or two checks of the same boolean as was the case before.